src/sage/arith/misc.py

```diff
@@ -2045,18 +2065,34 @@ def xgcd(a, b):
         sage: xgcd(y^2, a*h*y + b)
         (1, 7*a^2/b^2, (((-h)*a)/b^2)*y + 1/b)
     """
-    try:
+    if b is not None:
+        # xgcd of two elements
+        try:
+            return a.xgcd(b)
+        except AttributeError:
+            a = py_scalar_to_element(a)
+            b = py_scalar_to_element(b)
+        except TypeError:
+            b = py_scalar_to_element(b)
         return a.xgcd(b)
-    except AttributeError:
-        a = py_scalar_to_element(a)
-        b = py_scalar_to_element(b)
-    except TypeError:
-        b = py_scalar_to_element(b)
-    return a.xgcd(b)
+    else:
+        # xgcd of >=2 elements
+        if not isinstance(a, (tuple, list)):
+            raise TypeError("Input `a` has to be a tuple or a list.")
```

Comment:
```suggestion
            raise TypeError("input `a` should be a tuple or a list")
```
Errors should start with a lowercase alphabet, according to one of the hundreds of PEP I am too lazy to find. Also no punctuation at the end (IIRC)